packages/ag-charts-community/src/chart/dom/domManager.ts

```diff
@@ -13,38 +14,48 @@ const domElementConfig: Record<
     styles: { childElementType: 'style' },
     canvas: { childElementType: 'canvas' },
     'canvas-overlay': { childElementType: 'div' },
-    hidden: { childElementType: 'div' },
 };
 
 const STYLES = `
-.ag-chart-wrapper {
+.ag-charts-wrapper {
+    position: relative;
     touch-action: none;
     box-sizing: border-box;
-    overflow: hidden;
+    display: flex !important;
+    align-items: center !important;
+    justify-content: center !important;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
 }
 
-.ag-charts-style {
-    display: none;
+.ag-charts-canvas {
+    position: relative;
 }
 
-.ag-charts-hidden {
-    visibility: none;
+.ag-charts-canvas > * {
+    display: block;
 }
 
-.ag-charts-canvas, .ag-charts-canvas-overlay {
-    position: absolute;
-    display: block;
-    top: 0;
-    bottom: 0;
-    left: 0;
-    right: 0;
+.ag-charts-canvas-overlay {
+    position: relative;
+    display: contents;
```

Comment:
What's display: contents meant to be doing here?

Reply:
In one iteration it seemed that the `<div>` was adding some extra 3px to the canvas size (which caused an infinitely growing canvas feedback loop), this seemed to remove any extra contribution of this element from the impact on the parents size.